### PR TITLE
File_line checks provided after param if no match is found

### PR DESCRIPTION
**Pull request by bmjen**
This is a refactor of PRs: 
Closes https://github.com/puppetlabs/puppetlabs-stdlib/pull/427
Closes https://github.com/puppetlabs/puppetlabs-stdlib/pull/327
Closes https://github.com/puppetlabs/puppetlabs-stdlib/pull/317

Refactor eliminates the need to rewrite the file twice.

### CODE REVIEW

File: lib/puppet/provider/file_line/ruby.rb [hunk 1]
```diff
@@ -34,13 +34,22 @@ def lines
...
     match_count = count_matches(regex)
+
     if match_count > 1 && resource[:multiple].to_s != 'true'
      raise Puppet::Error, "More than one line in file '#{resource[:path]}' matches pattern '#{resource[:match]}'"
     end
+
     File.open(resource[:path], 'w') do |fh|
       lines.each do |l|
         fh.puts(regex.match(l) ? resource[:line] : l)
+        if (match_count == 0 and resource[:after])
```
Comment (hunner) [inline lib/puppet/provider/file_line/ruby.rb]: Want to use `regex_after` instead of `resource[:after]`?

Comment (bmjen) [inline lib/puppet/provider/file_line/ruby.rb]: They would both logically be the same, nil or ! nil, just a matter of readability and preference I guess.

File: spec/unit/puppet/provider/file_line/ruby_spec.rb [hunk 2]
```diff
@@ -140,7 +140,54 @@
...
+          let(:match) { '^foo2$' }
+          let(:after) { '^foo1$' }
+          let(:resource) {
+            Puppet::Type::File_line.new(
+              {
+                :name => 'foo',
+                :path => @tmpfile,
+                :line => 'inserted = line',
+                :after => after,
+                :match => match,
```
Comment (mhaskel) [inline spec/unit/puppet/provider/file_line/ruby_spec.rb]: I know this is nitpicky, but the misalignment hurts my eyes. :(

File: spec/unit/puppet/provider/file_line/ruby_spec.rb [hunk 3]
```diff
@@ -46,12 +46,12 @@
...
-           :name => 'foo',
-           :path => @tmpfile,
-           :line => 'foo = bar',
-           :match => '^foo\s*=.*$',
-          }
+        {
+          :name => 'foo',
+          :path => @tmpfile,
+          :line => 'foo = bar',
+          :match => '^foo\s*=.*$',
```
Comment (mhaskel) [inline spec/unit/puppet/provider/file_line/ruby_spec.rb]: also here

File: spec/unit/puppet/provider/file_line/ruby_spec.rb [hunk 4]
```diff
@@ -129,9 +129,9 @@
       let :resource do
         Puppet::Type::File_line.new(
           {
-            :name  => 'foo',
-            :path  => @tmpfile,
-            :line  => 'inserted = line',
+            :name => 'foo',
+            :path => @tmpfile,
+            :line => 'inserted = line',
```
Comment (mhaskel) [inline spec/unit/puppet/provider/file_line/ruby_spec.rb]: and here

**Pull request merged**